metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -1114,29 +1150,23 @@ def get_connection_objects(
         filter_pages = get_filter_pages(query_filter, page_size)
 
         for filter_page in filter_pages:
-            total_count = page_size
             has_next_page = 1
-            offset = 0
+            current_cursor: Optional[str] = None
             while has_next_page:
-                count = (
-                    page_size
-                    if offset + page_size < total_count
-                    else total_count - offset
-                )
+                filter_: str = make_filter(filter_page)
+
                 (
                     connection_objects,
-                    total_count,
+                    current_cursor,
                     has_next_page,
                 ) = self.get_connection_object_page(
-                    query,
-                    connection_type,
-                    make_filter(filter_page),
-                    count,
-                    offset,
+                    query=query,
+                    connection_type=connection_type,
+                    query_filter=filter_,
+                    fetch_size=self.config.fetch_size,
+                    current_cursor=current_cursor,
```

Comment:
this fetch logic is getting pretty difficult to follow

we have a page size, fetch size, cursor - can you give me a brief explanation of what's going on here?

Reply:
Added documentation.